### PR TITLE
chore(merge-tree sequence examples): remove NestBegin and NestEnd reference types

### PR DESCRIPTION
**Pull request by connorskees**
This is a ~conservative first pass at removing some of the tech debt related to the `NestBegin` and `NestEnd` reference types. Alongside these enum variants, this change removes `getStackContext`, `RangeStackMap`, the `shared-text` example, and a few other things. 

I will add some review comments to call out some changes (or lack thereof) to make sure they make sense once CI is green.

### CODE REVIEW

File: api-report/merge-tree.api.md [hunk 1]
```diff
@@ -539,9 +537,6 @@ export interface IMergeTreeTextHelper {
     getText(refSeq: number, clientId: number, placeholder: string, start?: number, end?: number): string;
 }
 
-// @public (undocumented)
-export function internedSpaces(n: number): string;
```
Comment (connorskees) [inline api-report/merge-tree.api.md:543]: Should this be deprecated in `main`? Do partners still consume `merge-tree`?

File: api-report/merge-tree.api.md [hunk 2]
```diff
@@ -860,9 +855,6 @@ export interface QProperty<TKey, TData> {
     key?: TKey;
 }
 
-// @public (undocumented)
-export type RangeStackMap = MapLike<Stack<ReferencePosition>>;
```
Comment (connorskees) [inline api-report/merge-tree.api.md:864]: Same question as above

File: api-report/merge-tree.api.md [hunk 3]
```diff
@@ -973,18 +961,9 @@ export enum ReferenceType {
     Transient = 256
 }
 
-// @public (undocumented)
-export const refGetRangeLabels: (refPos: ReferencePosition) => string[] | undefined;
```
Comment (connorskees) [inline api-report/merge-tree.api.md:977]: Same deprecation question

File: api-report/merge-tree.api.md [hunk 4]
```diff
@@ -973,18 +961,9 @@ export enum ReferenceType {
...
-
 // @public (undocumented)
 export const refGetTileLabels: (refPos: ReferencePosition) => string[] | undefined;
 
-// @public (undocumented)
-export function refHasRangeLabel(refPos: ReferencePosition, label: string): boolean;
-
-// @public (undocumented)
-export function refHasRangeLabels(refPos: ReferencePosition): boolean;
```
Comment (connorskees) [inline api-report/merge-tree.api.md:986]: Deprecation question

File: examples/data-objects/prosemirror/src/fluidBridge.ts
Comment (connorskees) [inline examples/data-objects/prosemirror/src/fluidBridge.ts:1]: For prosemirror, the example seems _good enough_ if we just do some finagling with the reference types. That said, it's not clear to me which reference type these should switch to, and the APIs being used appear quite old. I'm not sure if we want to remove this example entirely.

File: examples/data-objects/shared-text/.eslintrc.js
Comment (connorskees) [inline examples/data-objects/shared-text/.eslintrc.js:1]: shared-text uses _a lot_ of merge-tree internals, and it seems best just to delete entirely.

File: examples/data-objects/webflow/src/clipboard/paste.ts
Comment (connorskees) [inline examples/data-objects/webflow/src/clipboard/paste.ts:1]: Webflow seems _good enough_ if we just rip out some of the tags stuff.

File: packages/dds/merge-tree/src/localReference.ts [hunk 5]
```diff
@@ -354,7 +349,7 @@ export class LocalReferenceCollection {
 
 			lref.link(this.segment, offset, atRefs.push(lref).last);
 
-			if (refHasRangeLabels(lref) || refHasTileLabels(lref)) {
+			if (refHasTileLabels(lref)) {
 				this.hierRefCount++;
```
Comment (connorskees) [inline packages/dds/merge-tree/src/localReference.ts:353]: I kept `hierRefCount` and the tiles stuff around as I wasn't sure if this safe to remove.

File: packages/dds/merge-tree/src/mergeTree.ts [hunk 6]
```diff
@@ -383,34 +263,16 @@ function extendIfUndefined<T>(base: MapLike<T>, extension: MapLike<T> | undefine
...
 		this.rightmostTiles = createMap<ReferencePosition>();
 		this.leftmostTiles = createMap<ReferencePosition>();
-		this.rangeStacks = createMap<Stack<ReferencePosition>>();
 	}
 
 	public hierBlock() {
 		return this;
 	}
-
-	public hierToString(indentCount: number) {
```
Comment (connorskees) [inline packages/dds/merge-tree/src/mergeTree.ts:399]: This didn't seem to have any usage anywhere

File: packages/dds/sequence/src/test/testFarm.ts
Comment (connorskees) [inline packages/dds/sequence/src/test/testFarm.ts:1]: See also: https://github.com/microsoft/FluidFramework/pull/17547/commits/9dc1859c0c233bee932f36d1e31d762ba5f3f5c4

File: api-report/merge-tree.api.md (hunk 1 above)
Comment (anthony-murphy) [reply api-report/merge-tree.api.md:543]: they consume the package, but not the class. basically if it impacts the api-report or the type tests it should be deprecated in main before removal.

**Review by anthony-murphy — DISMISSED** [review]
we need to do some deprecations in main for completeness, but i highly doubt any of this is in use. we also need a changeset here to call out the removals in mergetree/seq, keep it brief, as again, i don't think its used, as this stuff probably doesn't work anyway.

File: examples/data-objects/webflow/src/clipboard/paste.ts
Comment (anthony-murphy) [reply examples/data-objects/webflow/src/clipboard/paste.ts:1]: i would like to keep web flow, but it is deleted here

Comment (connorskees) [reply examples/data-objects/webflow/src/clipboard/paste.ts:1]: See https://github.com/microsoft/FluidFramework/pull/17546/commits/61e4d8b1d1746306e770037b136c63ab338dd1e2

**Review by anthony-murphy — COMMENTED** [review]
actually, can we bring back web flow and test farm, feel free to rip features out, but i'd like to keep both. might make sense to do this with the deprecations in main

File: examples/data-objects/webflow/src/clipboard/paste.ts
Comment (connorskees) [reply examples/data-objects/webflow/src/clipboard/paste.ts:1]: ^ There were some weird test failures I couldn't reproduce locally: https://dev.azure.com/fluidframework/public/_build/results?buildId=195017&view=results

**Comment by connorskees** [comment]
testFarm wasn't deleted in this PR, it just has a lot of lines deleted. I _did_ delete it in https://github.com/microsoft/FluidFramework/pull/17547, however.

**Comment by connorskees** [comment]
Deprecation PR: https://github.com/microsoft/FluidFramework/pull/17555

File: packages/dds/sequence/src/intervals/sequenceInterval.ts [hunk 7]
```diff
@@ -577,8 +577,7 @@ export function createSequenceInterval(
 		endRefType = ReferenceType.Transient;
 	} else {
 		if (intervalType === IntervalType.Nest) {
```
Comment (anthony-murphy) [inline packages/dds/sequence/src/intervals/sequenceInterval.ts:579]: i think we should also deprecate IntervalType.Nest

**Review by anthony-murphy — APPROVED** [review]

**Pull request merged**